### PR TITLE
Replace Usage of `DispatchQueue` with `Task`

### PR DESCRIPTION
**Pull request by IsaacMarovitz**

### CODE REVIEW

File: PlayCover/AppInstaller/Installer.swift [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Installer {
 
         InstallVM.shared.next(.begin, 0.0, 0.0)
 
-        DispatchQueue.global(qos: .userInitiated).async {
+        Task {
```
Comment (KhoraLee) [inline PlayCover/AppInstaller/Installer.swift]: No need to add `.userInitiated` priority?

File: PlayCover/Model/PlayApp.swift [hunk 2]
```diff
@@ -86,7 +86,7 @@ class PlayApp: BaseApp {
                             &assertionID)
                         if success == kIOReturnSuccess {
                             while true { // Run a loop until the app closes
-                                Thread.sleep(forTimeInterval: 10) // Wait 10s
+                                try await Task.sleep(nanoseconds: 10000000000) // Sleep for 10 seconds
```
Comment (KhoraLee) [inline PlayCover/Model/PlayApp.swift:89]: `try await Task.sleep(for: .second(10))`

File: PlayCover/ViewModel/InstallVM.swift [hunk 3]
```diff
@@ -25,28 +26,20 @@ class InstallVM: ObservableObject {
...
                 self.installing = true
             } else if step == .finish || step == .failed {
                 self.progress = 1.0
-                DispatchQueue.global(qos: .userInteractive).async {
-                    usleep(1500000)
-                    DispatchQueue.main.async {
-                        self.installing = false
-                    }
-                }
+                try await Task.sleep(nanoseconds: 1500000000)
```
Comment (KhoraLee) [inline PlayCover/ViewModel/InstallVM.swift:36]: `try await Task.sleep(for: .second(1.5))`

File: PlayCover/ViewModel/InstallVM.swift [hunk 4]
```diff
@@ -25,28 +26,20 @@ class InstallVM: ObservableObject {
...
-            DispatchQueue.global(qos: .userInitiated).async {
-                while self.status == NSLocalizedString(step.rawValue, comment: "") {
-                    usleep(50000)
-                    DispatchQueue.main.async {
-                        if self.progress < stopProgress {
-                            self.progress += 0.002
-                        }
-                    }
+            while self.status == NSLocalizedString(step.rawValue, comment: "") {
+                try await Task.sleep(nanoseconds: 50000000)
```
Comment (KhoraLee) [inline PlayCover/ViewModel/InstallVM.swift:40]: `try await Task.sleep(for: .millisecond(50))`

File: PlayCover/Model/PlayApp.swift (hunk 2 above)
Comment (amirsaam) [reply PlayCover/Model/PlayApp.swift:89]: @KhoraLee  `sleep(for:)` is only available in macOS 13, as other suggestions.

Comment (IsaacMarovitz) [reply PlayCover/Model/PlayApp.swift:89]: As Amir said, `sleep(for:)` is only available on macOS 13+

File: PlayCover/ViewModel/NetworkVM.swift [hunk 5]
```diff
@@ -5,37 +5,6 @@
 //  Created by Isaac Marovitz on 09/10/2022.
 //
 
-// import Network
```
Comment (Depal1) [inline PlayCover/ViewModel/NetworkVM.swift:8]: This should be done in another PR.

**Review by Depal1 — CHANGES_REQUESTED** [review]
The rest looks good.

File: PlayCover/ViewModel/NetworkVM.swift (hunk 5 above)
Comment (IsaacMarovitz) [reply PlayCover/ViewModel/NetworkVM.swift:8]: This code never seems to work and contains the usage of DispatchQueue, hence my reason for deleting it.

Comment (Depal1) [reply PlayCover/ViewModel/NetworkVM.swift:8]: It is unrelated to the main changes of this PR.

Comment (Depal1) [reply PlayCover/ViewModel/NetworkVM.swift:8]: Either that or commit squashing.

**Review by Depal1 — APPROVED** [review]
